src/test/java/it/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPosterIT.java

```diff
@@ -0,0 +1,283 @@
+package it.com.atlassian.bitbucket.jenkins.internal.status;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketRepository;
+import com.atlassian.bitbucket.jenkins.internal.model.BuildState;
+import com.atlassian.bitbucket.jenkins.internal.status.BitbucketRevisionAction;
+import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder;
+import hudson.model.*;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.BitbucketJenkinsRule;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.BitbucketProxyRule;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.GitHelper;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.JenkinsProjectHandler;
+import jenkins.branch.MultiBranchProject;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import wiremock.org.apache.http.HttpStatus;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+import static com.atlassian.bitbucket.jenkins.internal.model.BuildState.SUCCESSFUL;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static it.com.atlassian.bitbucket.jenkins.internal.fixture.JenkinsProjectHandler.MASTER_BRANCH_PATTERN;
+import static it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import static java.lang.String.format;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Following tests uses real Bitbucket server and Jenkins for integration testing however the build status is posted against
+ * a stub. One of the primary reason is parallel development. Since we can only start *released* bitbucket server, we would
```

Comment:
```suggestion
 * a stub. One of the primary reasons is parallel development. Since we can only start a *released* bitbucket Server version, we would
```